V1.4 bugfix ug (#239)
AY2122S2-CS2103T-T09-2/tp: docs/UserGuide.md

* Prefixes can be in any order.<br>
  e.g. If the command specifies `n/NAME p/PHONE_NUMBER`, `p/PHONE_NUMBER n/NAME` is also acceptable.

> * If an input contains `"/"`, you cannot put spacing after its prefix.<br>

Reviewer: ```suggestion
* If an input contains `"/"`, you may wish to avoid putting spaces after `"/"`. This is to prevent the symbol from being recognized as a prefix.<br>
```

>   e.g. `s/Java/Python` will be interpreted as skills in `Java/Python` but if you specify

Reviewer: ```suggestion
  e.g. `s/Java/Python` will be interpreted as a single skill of `Java/Python` but if you specify
```

  `n/Bob s/  Java/Python`, `Java/` will be interpreted as a prefix.

* If an input is expected only once in the command but you specified it multiple times,
  only the last occurrence of the input will be taken.<br>
  e.g. If you specify `p/12341234 p/56785678`, only `p/56785678` will be taken.

> * If a valid prefix is given twice but you only specified input for the last occurrence of prefix, you will not receive errors.

Reviewer: I feel like if we claim 'you will not receive errors', there might be some groups flagging this as a false claim.

```suggestion
* If a valid prefix is given twice but you only specified the input for the last occurrence of said prefix, then the input will be accepted.
```

>   However, if you left the input for the last prefix empty, you will receive error message.<br>

Reviewer: ```suggestion
  However, if you left the input for the last prefix empty, you will receive an error message.<br>
```

>   e.g. If you specify `p/ p/56785678`, `p/56785678` will be taken but if you specify `p/56785678 p/`, you will receive error message.

Reviewer: Same as above, small grammar point to add 'an'

---
### Viewing a specific applicant : `view`

> Allows you to view on a specific applicant, by inputting the applicant's **full name**. 

Reviewer: I think without 'on' would make it more grammatically sound

```suggestion
Allows you to view a specific applicant, by inputting the applicant's **full name**. 
```

  e.g. If an applicant's full name is `Steve Jobs`, and the command provided is `view Steve` then LinkedOUT will not
  display `Steve Jobs`'s details. <br>
* The view command is case-insensitive. <br>
  e.g Viewing `hans` will match an applicant with name `Hans`.
* Only full name will be matched. <br>
  e.g. Viewing `Han Lee` will not match an applicant with name `Han`.

> * Only applicant with correct spacing in full name will be matched. <br>

Reviewer: I think would be good to give an example here as well.

```suggestion
* Only applicants with the right amount of spacing in their full name will be matched. <br>
e.g `Han       Lee` will not match `Han Lee`. <br>
```

---
### Searching for an applicant : `search`

> Allows you to search for applicants containing the input keywords and the keyword has to be an exact word. You can search for the applicant based on 

Reviewer: @zzhikai I think in our GUI pic at the start, would be good to label what is a keyword so that people who read aren't confused. Not sure if using keywords vs input will cause confusion.

> name, job, round or skills. You can also search for a combination of keywords and the applicant with the most matching keywords will be on the top of the list.

Reviewer: I think u can move this to the bottom part with multiple dashes, and give an example to illustrate.

<br>

Format:
```
search [n/NAME]…​ [j/JOB]…​ [r/ROUND]…​ [s/SKILL]…​
```
* You must provide a full match of keyword you want to search for.<br>

>   e.g. `Hans` will match `Hans Lee` but will not match `Han`. <br>

Reviewer: I think can be more descriptive.

```suggestion
  e.g. Searching `n/Hans` will match an applicant with name `Hans Lee` but will not match an applicant with name `Han`. <br>
```

* The search command is case-insensitive. <br>

>   e.g `hans` will match `Hans`.

Reviewer: Same as above, just a suggestion.

---
### Searching for an applicant : `search`

Allows you to search for applicants containing the input keywords and the keyword has to be an exact word. You can search for the applicant based on 
name, job, round or skills. 

<br>

Format:
```
search [n/NAME]…​ [j/JOB]…​ [r/ROUND]…​ [s/SKILL]…​
```
* You must provide a full match of keyword you want to search for.<br>
  e.g. Searching `n/Hans` will match an applicant with name `Hans Lee` but will not match an applicant with name `Han`. <br>
* The search command is case-insensitive. <br>
  e.g Searching `n/hans` will match an applicant with name `Hans`.
* You can search for a combination of keywords. 

>   After a search command, applicants will be displayed in descending order of matched keywords, with the most matched applicant on the top of the list.

Reviewer 2: Line 277: Consider changing to .... descending order of (number of) matched keywords, ~~with the most matched applicant on the top of the list.~~ with the applicant with the most number matches at the top of the list
